edb/pgsql/metaschema.py

```diff
@@ -1961,7 +1962,10 @@ class JSONIndexByTextFunction(dbops.Function):
                         'cannot index json '
                         || coalesce(jsonb_typeof(val), 'UNKNOWN')
                     ),
-                    detail => detail
+                    detail => (
+                        '{"hint":"Retrieving an element by an string index '
+                        || 'is only available for json arrays and objects."}'
```

Comment:
Why do we allow that for arrays?

Also please use all-caps spelling of JSON

Reply:
You are right, we don't allow this for arrays, we just have a more specialized error for them. Wording fixed.

I actually used lower-case `json` on purpose here because that is the name of the EdgeDB JSON scalar. But if you think that we should make it `JSON`, would that also apply to the main error message, such as `json index 10 is out of bounds`? There are a few of those, too.

Reply:
> I actually used lower-case json on purpose here because that is the name of the EdgeDB JSON scalar.

Then just spell it `std::json`?

Reply:
After reviewing the error messages themselves I came to the conclusion that using `JSON` is more appropriate because a lot of the messages refer to `JSON string` or `JSON object` or `JSON array`, which are various valid concepts, but are not distinct types. Also the ambiguous meaning of `std::json array` would be unhelpful (is it `array<json>` or `json` that happens to be an array internally?)